src/common/AriaQueryHandler.ts

```diff
@@ -111,6 +121,29 @@ const queryAllArray = async (
   return jsHandle;
 };
 
+// If multiple waitFor are set up asynchronously, we need to wait for the first
+// one to set up the binding in the page before running the others.
+let settingUpBinding = null;
+
+async function addHandlerToWorld(domWorld: DOMWorld) {
+  if (settingUpBinding) {
+    await settingUpBinding;
+  }
+  if (!domWorld.hasBinding('ariaQuerySelector')) {
+    let done: () => void | null = null;
+    settingUpBinding = new Promise((resolve) => {
+      done = resolve;
+    });
+    await domWorld.addBinding('ariaQuerySelector', async (selector: string) => {
```

Comment:
Can you do `settingUpBinding  = domWorld.addBinding(...)` and then `await settingUpBinding`? I think it might help avoid having the extra promise?

Reply:
Done, but moved into DOMWorld instead.